cranelift/isle/isle_examples/conversions.isle

```diff
@@ -0,0 +1,12 @@
+(type T (enum (A) (B)))
```

Comment:
This is unrelated to these particular changes, but long overdue nonetheless: can we write a `#[test]` that asserts that every file in `isle_examples` compiles okay? (At least through ISLE, testing the emitted code with `rustc` will be a bit harder/more tedious since it requires glue code).

And for the ones that are supposed to fail (I forget if we even have any of those) it would be nice if we had a comment at the top with a substring of the expected error message. These could be in a separate directory and also be handled in separate PRs.

But we really need *some* kind of testing story for this stuff, especially as we tweak the language and compiler.

Reply:
Yes, definitely agreed, this is probably as good a time as any to start!

Reply:
Done! We now have pass-tests, fail-tests, and link-tests; the last one checks that the generated code builds and links against the example driver/context trait impl.